### DISCUSSION TITLE
Onix - harp synchronisation with adaptor board

### DISCUSSION THREAD

**Discussion by ablot** [harp-tech/protocol#88] [Q&A]
Hi all,

I received a tiny adaptor pcb for synchronising my Onix recordings with my harp device and I had a few questions

1 - The Harp Timestamp Generator input to the PCB must come from a harp clock sync device or can any device be used? (can the clkin of a behaviour device also be used for generating timestamp for instance?)
2 - If I have a sync device and a behaviour device, what is the difference between connecting the behaviour directly to an output of the clock sync vs connecting it to the "another harp device" output of the tiny pcb adaptor?
3 - On the bonsai-onix side, how are things working? Do I just need to save the output of a HARPSyncInputDevice or are there other functionalities?

**Comment by bruno-f-cruz** [comment]
1 - As defined in the current [harp protocol specs](https://github.com/harp-tech/protocol/blob/main/Device.md) you have two types of devices: those that are able to generate a clock-synch and those that are not. Only the former can be used as sources of sychronization. Currently, only the [Device.ClockSynchronizer](https://github.com/harp-tech/device.clocksynchronizer/tree/main) and [Device.TimestampGenerator3](https://github.com/harp-tech/device.timestampgeneratorgen3/tree/main) can function as sources. All other devices can simply be slaved to this clock signal.

2- See 1), I believe it makes this point clear

3 - I dont see this any reference to `HarpSyncInputDevice` in the harp-tech github organization, not sure what this is.

**Comment by ablot** [comment]
1- Thanks. That makes things clear and simple.

2- I'm still a bit unsure about this point. The board I have has a clkout (the "another harp device" in this picture if I understood correctly)
![image](https://github.com/harp-tech/harp-tech.github.io/assets/2701493/248f7c09-6bf0-4909-a1ec-60b7d4a06c19)
Is that clkout just a simple copy of the signal that came in the clkin input?

3- That's part of Bonsai.Onix I'll redirect the question there.

**Reply by bruno-f-cruz** [reply]
Ok 2- relates to 3-. As I said before, in the harp-tech organization I don't see any info related to that device, so I assume that you got it from open-ephys? Might be better to just ask them directly. Thanks!
